### DISCUSSION TITLE
Git-lfs: emulating push command

### DISCUSSION THREAD

**Discussion by Smith1123** [git-lfs/git-lfs#4941] [Q&A] [answered]
Hey Dear Git-lfs Developers,
I'm using git-lfs on my project, and I use my own nexus server to store the files. I would like to use it in a stateless way, but unfortunately, I couldn't reinitialise git-lfs repository on my nexus server, because it isn't implemented any upload API to this type of repository. Could you please provide me a protocol or just the curl requests example that you guys use to upload contents to nexus? I looked into nexus logs, and I found out, it is sent 3 requests. (POST,PUT,POST in that order), the first one I don't know what is it, I think the PUT is the upload, and the last one is to verify the content. Help me with this request will be really appreciated  :)

Thank you a lot

Best,
Zsolt Fehér

**Comment by bk2204** [comment, marked as answer]
Hey,

We absolutely have documentation for the API, and you can see it right here: https://github.com/git-lfs/git-lfs/tree/main/docs/api.  That should cover all of the HTTPS endpoints required.

Feel free to let us know if you still have questions.

**Reply by Smith1123** [reply]
Thank you, I saw this page earlier, but somehow I slipped over the batch api. Anyway it is working fine now :)
